### PR TITLE
cmd/snap: create SNAP_USER_DATA and common dirs in `snap run`

### PR DESCRIPTION
**Pull request by mvo5**
This is build on top of https://github.com/snapcore/snapd/pull/1247 and it avoids the need for the launcher to create the user data dirs. Instead we do it here and make the launcher as tiny as possible.

### CODE REVIEW

**Comment by mvo5** [comment]
retest this please

File: cmd/snap/cmd_run.go [hunk 1]
```diff
@@ -113,7 +116,25 @@ func snapExecAppEnv(app *snap.AppInfo) []string {
...
 }
 
+func createUserDataDirs(info *snap.Info) error {
+	usr, err := UserCurrent()
+	if err != nil {
+		return fmt.Errorf("cannot get the current user: %s", err)
+	}
+
+	// see snapenv.User
+	userData := filepath.Join(usr.HomeDir, info.MountDir())
```
Comment (zyga) [inline cmd/snap/cmd_run.go]: I love how we can do this reliably since we run as the user. +1

**Comment by zyga** [comment]
LGTM

File: cmd/snap/cmd_run.go [hunk 2]
```diff
@@ -99,6 +120,10 @@ func snapRun(snapApp, command string, args []string) error {
 		return fmt.Errorf("cannot find app %q in %q", appName, snapName)
 	}
 
+	if err := createUserDataDirs(info); err != nil {
+		logger.Noticef("WARNING: cannot create user data: %s", err)
```
Comment (niemeyer) [inline cmd/snap/cmd_run.go]: s/data/data directory/

File: cmd/snap/cmd_run.go [hunk 3]
```diff
@@ -85,7 +88,25 @@ func snapExecAppEnv(app *snap.AppInfo) []string {
...
+	for _, d := range []string{userData, commonUserData} {
+		if err := os.MkdirAll(d, 0755); err != nil {
+			return fmt.Errorf("cannot create %q: %s", d, err)
+		}
+	}
+	return nil
+}
+
 var SyscallExec = syscall.Exec
+var UserCurrent = user.Current
```
Comment (niemeyer) [inline cmd/snap/cmd_run.go]: These should go into export_test.go as well. Their only reason of being here is tests, I think.

**Comment by niemeyer** [comment]
Couple of trivials, and LGTM.

**Comment by mvo5** [comment]
retest this please

**Pull request merged**